token/core/zkatdlog/nogh/driver/base.go

```diff
@@ -81,12 +84,37 @@ func (d *base) newWalletService(
 		deserializerManager,
 		ignoreRemote,
 	)
+	// owner role
+	// we have one key manager for fabtoken and one for each idemix issuer public key
+	var kmps []common2.KeyManagerProvider
+	for _, key := range pp.IdemixIssuerPublicKeys {
```

Comment:
Is there a callback when we change pp, in order to create a new key manager and remove the ones that are not relevant anymore?

Reply:
yes, when the public params change a new TMS is loaded and together with it a new wallet service that reloads all the keys.